### PR TITLE
Improve GeanyProxyProbeResults a litte

### PR DESCRIPTION
**Pull request by codebrainz**
* Add GEANY_ prefix to enumerators.
  * Use a discrete set of enumerators so the proxy doesn't have to worry about flag bitmasks, which isn't required.
  * Update documentation.
  * Use the new enumerators in Geany.
  * Improve debug output for bogus values a bit.
  * Bump API version to 229.

### CODE REVIEW

File: src/plugindata.h [hunk 1]
```diff
@@ -379,7 +392,7 @@ GeanyProxyProbeResults;
 struct GeanyProxyFuncs
 {
 	/** Called to determine whether the proxy is truly responsible for the requested plugin.
-	 * A NULL pointer assumes the probe() function would always return @ref PROXY_MATCHED */
+	 * A NULL pointer assumes the probe() function would always return @ref GEANY_PROXY_MATCH */
 	gint		(*probe)     (GeanyPlugin *proxy, const gchar *filename, gpointer pdata);
```
Comment (b4n) [inline src/plugindata.h]: ideally the function would return `GeanyProxyProbeResults`, but well, it'd technically be an ABI break -- even though we know everyone use 2's complement so it wouldn't change nothing, but meh.

**Comment by b4n** [comment]
Makes sense.  The only advantage of flags over enumeration is to be able to combine them, but here it doesn't seem to really make any sense, and I can't easily think of a future change reusing `NOLOAD` with something else than `MATCH`.  And yeah, it's kind of unexpected to not be allowed to return `IGNORE | NOLOAD` yet them being flags.

File: src/plugindata.h (hunk 1 above)
Comment (kugel-) [inline src/plugindata.h]: I find NOLOAD is the better name

File: doc/plugins.dox [hunk 2]
```diff
@@ -807,17 +806,17 @@ Consider the 5 basic cases:
 1) A candidate comes with a suitable file extension but is not a workable plugin file at all. For
 example, your proxy supports plugins written in a shell script (.sh) but the shebang of that script
 points to an incompatible shell (or even lacks a shebang). You should check for this in
-GeanyProxyFuncs::probe() and return @ref PROXY_IGNORED which hides that script from the Plugin
+GeanyProxyFuncs::probe() and return @ref GEANY_PROXY_IGNORE which hides that script from the Plugin
 Manager and allows other enabled proxy plugins to pick it up. GeanyProxyFuncs::probe() returning
-@ref PROXY_IGNORED is an indication that the candidate is meant for another proxy, or the user
+@ref GEANY_PROXY_IGNORE is an indication that the candidate is meant for another proxy, or the user
```
Comment (kugel-) [inline doc/plugins.dox]: PROXY_IGNORE doesn't exist (was renamed to NOLOAD), can you please fix that also?

**Comment by kugel-** [comment]
Can we find a better name than _RELATED? IMO it doesn't suggest that it wont be loaded.

**Comment by codebrainz** [comment]
> Can we find a better name than _RELATED? IMO it doesn't suggest that it wont be loaded.

Correct, it suggest that it _may_ be loaded (by the proxy) and is related to a "matched" one, but is not matched itself. It makes sense from the proxy plugin's point of view. If you look at it from Geany's point of view, then NOLOAD makes some kind of sense, but this is the proxy plugin API so it should be from that perspective, IMO.

File: doc/plugins.dox (hunk 2 above)
Comment (codebrainz) [inline doc/plugins.dox]: What? It's [still in the enum](https://github.com/geany/geany/blob/master/src/plugindata.h#L356) and [documented there](https://github.com/geany/geany/blob/master/src/plugindata.h#L356) and in the [enum doc-comment](https://github.com/geany/geany/blob/master/src/plugindata.h#L344). I don't understand.

Comment (kugel-) [inline doc/plugins.dox]: Nevermind, I was confused.

**Comment by codebrainz** [comment]
> Can we find a better name than _RELATED

Can add later as a synonym without break ABI.

**Pull request merged**